src/Microsoft.OData.Core/Uri/ODataUriConversionUtils.cs

```diff
@@ -625,6 +625,7 @@ private static object ConvertFromResourceOrCollectionValue(string value, IEdmMod
         {
             ODataMessageReaderSettings settings = new ODataMessageReaderSettings();
             settings.Validations &= ~ValidationKinds.ThrowOnUndeclaredPropertyForNonOpenType;
+            settings.ReadUntypedAsString = false;

```

Comment:
We should retrieve it from the message reader setting. Should we? or add a configuration in the Uri query parsing?

Reply:
+1

Reply:
This setting controls how we read untyped values -- if our logic is to read untyped values as structured, not as string (which it should be) then I think it's correct to statically set to false.